src/components/Review/DecisionPreview/ReviewPreviewModal.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from 'react'
+import { Modal, Message, Button, Loader, ModalProps } from 'semantic-ui-react'
+import { useLanguageProvider } from '../../../contexts/Localisation'
+import { postRequest } from '../../../utils/helpers/fetchMethods'
+import config from '../../../config'
+import { getItemDisplayComponent, ActionResultPreviewData } from './PreviewItems'
+
+interface PreviewProps extends ModalProps {
+  open: boolean
+  setOpen: (open: boolean) => void
+  decision: string
+  reviewId: number
+  previewData: { [key: string]: any }
+}
+
+const previewEndpoint = `${config.serverREST}/preview-actions`
+
+const ReviewPreviewModal: React.FC<PreviewProps> = ({
+  open,
+  setOpen,
+  decision,
+  reviewId,
+  previewData,
+}) => {
+  const { strings } = useLanguageProvider()
+  const [data, setData] = useState<ActionResultPreviewData[] | null>(null)
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(false)
+
+  useEffect(() => {
+    if (!open) {
+      setData(null)
+      setError(false)
+      return
```

Comment:
Reset state when closing modal, otherwise it's still there next time you open it.